deep_qa/models/sentence_selection/siamese_sentence_selector.py

```diff
@@ -0,0 +1,115 @@
+        batch to get a probability distribution over sentences in the passage.
+        """
+        # First we create input layers and pass the inputs through embedding layers.
+        # shape: (batch size, question_length)
+        question_input = Input(shape=self._get_sentence_shape(self.max_question_length),
+                               dtype='int32', name="question_input")
+
+        # shape: (batch size, num_sentences, sentences_length)
+        sentences_input_shape = ((self.num_sentences,) +
+                                 self._get_sentence_shape(self.max_sentence_length))
+        sentences_input = Input(shape=sentences_input_shape,
+                                dtype='int32', name="sentences_input")
+
+        # shape: (batch size, question_length, embedding size)
+        question_embedding = self._embed_input(question_input)
+
+        # shape: (batch size, num_sentences, sentences_length, embedding size)
+        sentences_embedding = self._embed_input(sentences_input)
+
+        # We encode the question embeddings with some encoder.
+        question_encoder = self._get_encoder(name="question_encoder")
+        # shape: (batch size, encoder_output_dimension)
+        encoded_question = question_encoder(question_embedding)
+
+        # We encode the document embeddings with some encoder.
+        sentences_encoder = EncoderWrapper(self._get_encoder(name="sentence_encoder"),
+                                           name="TimeDistributed_sentences_encoder")
+        # shape: (batch size, num_sentences, encoder_output_dimension)
+        encoded_sentences = sentences_encoder(sentences_embedding)
+
+        # Here we use the Attention layer with the cosine similarity function
+        # to get the cosine similarities of each sesntence with the question.
+        # shape: (batch size, num_sentences)
+        attention_name = 'question_sentences_similarity'
+        similarity_params = {"type": "cosine_similarity"}
+        sentence_probabilities = Attention(name=attention_name,
+                                           similarity_function=similarity_params)([encoded_question,
+                                                                                   encoded_sentences])
+
+        return DeepQaModel(input=[question_input, sentences_input],
+                           output=sentence_probabilities)
+
+    @overrides
+    def _instance_type(self):
+        """
+        Return the instance type that the model trains on.
+        """
+        return SentenceSelectionInstance
+
+    @overrides
+    def _get_max_lengths(self) -> Dict[str, int]:
+        """
+        Return a dictionary with the appropriate padding lengths.
+        """
+        max_lengths = super(SiameseSentenceSelector, self)._get_max_lengths()
+        max_lengths['num_question_words'] = self.max_question_length
+        max_lengths['num_sentences'] = self.num_sentences
+        max_lengths['num_sentence_words'] = self.max_sentence_length
```

Comment:
Because `num_sentence_words` is actually a key handled by the super class, you don't need it here, or anywhere else in this class.